FastSimulation/Tracking/src/SeedFinderSelector.cc

```diff
@@ -119,8 +153,108 @@ bool SeedFinderSelector::pass(const std::vector<const FastTrackerRecHit *>& hits
 	    return !tripletresult.empty();
 	}
+	else if(CAHitTriplGenerator_)
+	{  
+	  return true;
+	}
+    }
 
+    if(CAHitQuadGenerator_)
+    {
+      std::cout<<"Inside CAHit section"<<std::endl;
+      if(hits.size() < 4)
+	{
+	  throw cms::Exception("FastSimTracking") << "For the given configuration, SeedFinderSelector::pass requires at least 4 hits";
+        }
+
+      if(!seedingLayer)
+	std::cout<<"SeedingLayers pointer not set"<<std::endl;
+
+      SeedingLayerSetsHits & layers = *seedingLayer;
+
+      for(int i=0; i<(int)hits.size()-1; i++){
+        //-----------------determining hit layer---------------                                                                                                                
+	std::string hitlayer[2] = {};
+        int layerNo = -1;
+	std::string side;
+        bool IsPixB = false;
+	SeedingLayerSetsHits::SeedingLayerSet pairCandidate;
+        //hit 1                                                                                                                                                                
+        if( (hits[i]->det()->geographicalId()).subdetId() == PixelSubdetector::PixelBarrel){
+          layerNo = (*trackerTopology.product()).pxbLayer(hits[i]->det()->geographicalId());
+          IsPixB = true;
+        }
+        else if ((hits[i]->det()->geographicalId()).subdetId() == PixelSubdetector::PixelEndcap){
+          layerNo = (*trackerTopology.product()).pxfDisk(hits[i]->det()->geographicalId());
+          side = (*trackerTopology.product()).pxfSide(hits[i]->det()->geographicalId())==1 ? "_neg" : "_pos";
+          IsPixB = false;
+        }
+        hitlayer[0] = LayerName(layerNo, side, IsPixB);
+        //hit 2                                                                                                                                                                
+        if( (hits[i+1]->det()->geographicalId()).subdetId() == PixelSubdetector::PixelBarrel){
+          layerNo = (*trackerTopology.product()).pxbLayer(hits[i+1]->det()->geographicalId());
+          IsPixB = true;
+        }
+        else if ((hits[i+1]->det()->geographicalId()).subdetId() == PixelSubdetector::PixelEndcap){
+          layerNo = (*trackerTopology.product()).pxfDisk(hits[i+1]->det()->geographicalId());
+          side = (*trackerTopology.product()).pxfSide(hits[i+1]->det()->geographicalId())==1 ? "_neg" : "_pos";
+          IsPixB = false;
+        }
+	hitlayer[1] = LayerName(layerNo, side, IsPixB);
+        for(SeedingLayerSetsHits::SeedingLayerSet ls : *seedingLayer){
+          for(const auto p : layerPairs_){
+            pairCandidate = ls.slice(p,p+2);
+	    std::string layerPair[2] = {};
+            int i=0;
+            for(auto layerSet : pairCandidate){
+              layerPair[i] = layerSet.name();
+              i++;
+            }
+            if((layerPair[0] == hitlayer[0] && layerPair[1] == hitlayer[1]))
```

Comment:
I have created a std::vector<SeedingLayerSetsBuilder::SeedingLayerId> for both the hit pairs and the layer pairs for the comparison to avoid using string. I am not sure if that was what you wanted.